src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -61,13 +78,25 @@ export const TransactionContract = ({
         fee={parseRelevantAssetFromTx(txDetails, AssetTypes.Fee)}
         explorerTxLink={txDetails.explorerTxLink}
         txid={txDetails.tx.txid}
+        txData={txMetadata}
         showDateAndGuide={showDateAndGuide}
         isFirstAssetOutgoing={isFirstAssetOutgoing}
         parentWidth={parentWidth}
       />
       <TransactionDetailsContainer isOpen={isOpen} compactMode={compactMode}>
         <Transfers compactMode={compactMode} transfers={txDetails.tx.transfers} />
         <TxGrid compactMode={compactMode}>
+          {txDetails.direction === Direction.InPlace &&
+          txMetadata &&
+          txMetadata?.assetId &&
+          txMetadata?.value ? (
```

Comment:
I'd love to see this as a `boolean` constant on line 67 or so to quickly grok what it does.

`isApprovalTransaction`, I assume.

Reply:
Unfortunately we would lose type narrowing by having such expression/function and would still have to implement redundant checks.
Same thing with a function, since TS uses control-flow analysis, this wouldn't work unless said function is a type guard, but lib does not export everything needed to make it happen currently.

Added a TODO for it in e0bea29ff9e9dc2dbe51502ebc4809669d8f9909